src/python/pants/backend/codegen/grpcio/python/grpcio_run.py

```diff
@@ -45,6 +47,16 @@ def execute_codegen(self, target, target_workdir):
       if exit_code != 0:
         raise TaskError('{} ... exited non-zero ({}).'.format(cmdline, exit_code),
                         exit_code=exit_code)
+      # Create __init__.py in each subdirectory of the target
+      # directory so that setup_py recognizes them as modules.
+      def iter_module_dirs():
+        yield target_workdir
+        for root, dirs, _ in safe_walk(target_workdir):
+          for dirname in dirs:
+            yield os.path.join(root, dirname)
+      for module_dir in iter_module_dirs():
+        with open(os.path.join(module_dir, '__init__.py'), 'a'):
+          pass
```

Comment:
We can use here a utility we have to identify missing `__init__.py` files: 

https://github.com/pantsbuild/pants/blob/c26f374caadc91de631a98ff540854ff923d2a2d/src/python/pants/backend/python/subsystems/pex_build_util.py#L79-L92

The function works like this:
```
$ ./pants repl src/python/pants/backend/python/subsystems

>>> from pants.backend.python.subsystems.pex_build_util import identify_missing_init_files
>>> identify_missing_init_files(["hello/world/test.py"])
{'hello/__init__.py', 'hello/world/__init__.py'}
```

So, here you would call  `identify_missing_init_files([target_workdir])`, then iterate over the result to create an empty `__init__.py` file.

--

In addition to the above change, it would be great to use `pathlib.Path` to simplify the creation of the file. Specifically, you could do this:

```python
for missing_init in identify_missing_init_files([target_workdir]):
  Path(missing_init).touch()
```

See https://docs.python.org/3/library/pathlib.html#pathlib.Path.touch.